products/BellHybrid/alarms/src/actions/FrontlightAction.cpp

```diff
@@ -54,17 +54,31 @@ namespace alarms
         }
     } // namespace
 
-    FrontlightAction::FrontlightAction(sys::Service &service, Mode mode)
-        : pimpl{createFrontlightImplementation(service, mode)}, settings{
-                                                                    service::ServiceProxy{service.weak_from_this()}}
+    FrontlightAction::FrontlightAction(sys::Service &service, Mode mode, SettingsDependency settingsDependency)
+        : settingsDependency{settingsDependency}, pimpl{createFrontlightImplementation(service, mode)},
+          settings{service::ServiceProxy{service.weak_from_this()}}
     {}
 
     bool FrontlightAction::execute()
     {
-        const std::string flontlightEnabled =
-            settings.getValue(bell::settings::Alarm::lightActive, settings::SettingsScope::Global);
-        if (flontlightEnabled == std::string(frontlightOFF)) {
-            return true;
+        std::string settingString;
+
+        switch (settingsDependency) {
+        case SettingsDependency::AlarmClock:
+            settingString = settings.getValue(bell::settings::Alarm::lightActive, settings::SettingsScope::Global);
+            if (settingString == std::string(alarmFrontlightOFF)) {
+                return true;
```

Comment:
Why not use different implementations (reuse the pimpl idiom which is already implemented) instead of introducing such conditions?
Is AlarmClock using both Linear and Manual modes, depending on the situation??

Reply:
@piotr-tanski 
I did not want to bind two different traits of an alarm:
- value changing characteristics (Manual/Linear)
and 
- configuration dependency
into one place. I did not want to create multiple frontligt actions differing only on condition check.
Im open to change, not sure if you wanted the explaination or require change here, please specify :)

Reply:
Please, keep in mind that the LinearProgressFrontlightAction is actually pre-wake up only. It uses pre-wake up settings like pre wake up light duration.
So actually LinearProgressFrontlightAction is pre wake up only, and the Manual one is alarm clock only in the current implementation. Probably this code is not needed in the common part then.